third_party/maya/lib/usdMaya/shadingModeExporterContext.h

```diff
@@ -0,0 +1,93 @@
+//    the License and to reproduce the content of the NOTICE file.
+//
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
+#ifndef PXRUSDMAYA_SHADINGMODEEXPORTERCONTEXT_H
+#define PXRUSDMAYA_SHADINGMODEEXPORTERCONTEXT_H
+
+#include "pxr/pxr.h"
+#include "usdMaya/util.h"
+#include "pxr/usd/usd/stage.h"
+
+
+#include <maya/MObject.h>
+#include <boost/function.hpp>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+
+class PxrUsdMayaShadingModeExportContext
+{
+public:
+    MObject GetShadingEngine() const { return _shadingEngine; }
+    const UsdStageRefPtr& GetUsdStage() const { return _stage; }
+
+    MObject GetSurfaceShader() const;
+
+    /// An assignment contains a bound prim path and a list of faceIndices. 
+    /// If the list of faceIndices is non-empty, then it is a partial assignment
+    /// targeting a subset of the bound prim's faces. 
+    /// If the list of faceIndices is empty, it means the assignment targets 
+    /// all the faces in the bound prim or the entire bound prim.
+    typedef std::pair<SdfPath, VtIntArray> Assignment;
+    
+    /// Vector of assignments.
+    typedef std::vector<Assignment> AssignmentVector;
+
+    /// Returns a vector of binding assignments associated with the shading 
+    /// engine.
+    AssignmentVector GetAssignments() const;
+
+    /// Use this function to create a UsdShadeMaterial prim at the "standard"
+    /// location.  The "standard" location may change depending on arguments
+    /// that are passed to the export script.
+    UsdPrim MakeStandardMaterialPrim(
+            const AssignmentVector& assignmentsToBind,
+            const std::string& name=std::string()) const;
+
+    /// Use this function to get a "standard" usd attr name for \p attrPlug.
+    /// The definition of "standard" may depend on arguments passed to the
+    /// script (i.e. stripping namespaces, etc.).
+    std::string GetStandardAttrName(const MPlug& attrPlug) const;
+
+    PxrUsdMayaShadingModeExportContext(
+            const MObject& shadingEngine,
+            const UsdStageRefPtr& stage,
+            bool mergeTransformAndShape,
+            const PxrUsdMayaUtil::ShapeSet& bindableRoots,
+            SdfPath overrideRootPath);
+private:
+    MObject _shadingEngine;
+    const UsdStageRefPtr& _stage;
+    bool _mergeTransformAndShape;
+    bool _handleUsdNamespaces;
+    SdfPath _overrideRootPath;
+
+    SdfPathSet _bindableRoots;
+};
+
+typedef boost::function< void (PxrUsdMayaShadingModeExportContext*) > PxrUsdMayaShadingModeExporterContext;
```

Comment:
From what we could tell, this typedef isn't used anywhere -- could you confirm and remove if it isn't?

Otherwise, (and this is nit-picky) -- could we rename this typedef (and use a C++11 type alias), since this type isn't really a context, it's a function accepting a context?